File: app/src/main/java/com/pranav/java/ide/ui/treeview/model/TreeFolder.java
package com.pranav.java.ide.ui.treeview.model;

import android.content.Context;
import android.graphics.drawable.Drawable;

import androidx.appcompat.content.res.AppCompatResources;

import com.pranav.java.ide.R;

import java.io.File;

public class TreeFolder extends com.pranav.java.ide.ui.treeview.file.TreeFile {

    public TreeFolder(File file) {
        super(file);
    }

    @Override
    public Drawable getIcon(Context context) {
        return AppCompatResources.getDrawable(context, R.drawable.folder);
    }
}

File: app/src/main/java/com/pranav/java/ide/ui/treeview/model/TreeFolder.kt
package com.pranav.java.ide.ui.treeview.model

import android.content.Context
import android.graphics.drawable.Drawable

import androidx.appcompat.content.res.AppCompatResources

import com.pranav.java.ide.R
import com.pranav.java.ide.ui.treeview.file.TreeFile

import java.io.File

class TreeFolder(file: File) : TreeFile() {

    init {
        super(file)
    }

    override fun getIcon(context: Context): Drawable {
        return AppCompatResources.getDrawable(context, R.drawable.folder)
    }
}
